app/ante/eth.go

```diff
@@ -244,28 +260,25 @@ func CheckEthCanTransfer(
 		if value == nil || value.Sign() == -1 {
 			return fmt.Errorf("value (%s) must be positive", value)
 		}
-		from := common.BytesToAddress(msgEthTx.From)
 		// check that caller has enough balance to cover asset transfer for **topmost** call
 		// NOTE: here the gas consumed is from the context with the infinite gas meter
-		if value.Sign() > 0 && !canTransfer(ctx, evmKeeper, evmParams.EvmDenom, from, value) {
-			return errorsmod.Wrapf(
-				errortypes.ErrInsufficientFunds,
-				"failed to transfer %s from address %s using the EVM block context transfer function",
-				value,
-				from,
-			)
+		if value.Sign() > 0 {
+			balance := balanceGetter(msgEthTx.GetFrom(), evmParams.EvmDenom)
+			if balance.Cmp(value) < 0 {
+				from := common.BytesToAddress(msgEthTx.From)
+				return errorsmod.Wrapf(
+					errortypes.ErrInsufficientFunds,
+					"failed to transfer %s from address %s using the EVM block context transfer function",
+					value,
+					from,
+				)
+			}
 		}
 	}
 
 	return nil
 }
 
-// canTransfer adapted the core.CanTransfer from go-ethereum
-func canTransfer(ctx sdk.Context, evmKeeper EVMKeeper, denom string, from common.Address, amount *big.Int) bool {
-	balance := evmKeeper.GetBalance(ctx, sdk.AccAddress(from.Bytes()), denom)
-	return balance.Cmp(amount) >= 0
```

Comment:
balance could be nil, but there was no nil check before

Reply:
try to reproduce in integration test?